docs/features.rst

```diff
@@ -121,6 +121,33 @@ attribute, matching regular expressions are replaced by an empty
 string. For dictionaries it's possible to provide list of keys
 which should be removed.
 
+Substitution of current values can be done by appending a ``~`` suffix
+to the key name. The pattern and replacement parameters need to be provided
+as values in the form of ``<d>PATTERN<d>REPLACEMENT<d>``, where ``<d>`` is delimiter
+which can be any character however such character cannot be then used withing PATTERN
+and REPLACEMENT text as escaping isn't supported. These input can be either a string or
+list of strings.
+
+The `re.sub`__ is used to do the substitution thus all feaures of ``re.Pattern`` can
+be used (named groups, backreferencing...).
+
+In the fmf file it is better to use single quotes ``'`` as they do not need such intensive escaping::
+
+    require~: ';^foo;foo-ng;'
```

Comment:
Just a quick first-sight brainstorm: Instead of describing the delimiters and parsing them and escaping them wouldn't it be easier to have two keys?

```yaml
require~:
    pattern: foo
    replace: foo-ng
```

Or do you find to too long?

Reply:
@psss Some details in https://github.com/teemtee/tmt/discussions/2430#discussioncomment-9131018 

I think that fmf prefers shorter way of syntax so "sed"-like seems (IMO) to be a better fit.

Reply:
It could be confusing because no other operation sets dict keys. I am thinking of how it fits along with the `+` operator where the value is the operand. It's hard to describe

Reply:
Thanks for the pointer. I've read he discussion. Makes sense to keep the syntax concise.